include/string_utils.h

```diff
@@ -244,7 +244,17 @@ constexpr bool iequals(T1&& a, T2&& b)
 	return std::equal(str_a.begin(), str_a.end(), str_b.begin(), str_b.end(), ciequals);
 }
 
-char *strip_word(char *&cmd);
+// Performs a "natural" comparison between A and B, which is case-insensitive
+// and treats number sequenences as whole numbers. Returns true if A < B. This
+// function can be used with higher order sort rountines, like std::sort.
+//
+// Examples:
+// - ("abc_2", "ABC_10") -> true, because abc_ matches and 2 < 10.
+// - ("xyz_2", "ABC_10") -> false, because 'x' > 'a'.
+// - ("abc123", "abc123=") -> true, simply because the first is shorter.
```

Comment:
So what about `("Z", "abc123=")`? In this case you'd need to fall back to the default alphanumeric sort, I think.

I think how it should work is as follows:

- If the two strings have the same prefix (without taking casing into account) and they in numbers, then do the natural sort thing
- In any other scenario, default to alphanumeric sort

Reply:
> If the two strings have the same prefix (without taking casing into account) and they in numbers, then do the natural sort thing
> In any other scenario, default to alphanumeric sort

:+1: that's how it works.

Here's a fun (and contrived!) example of a studio in the middle of development. 

They've got different people working on the intro, game data, ending sequence (currently two drafts remaining), and outro.

Each team shares the floppy disk images:

![2023-02-09_10-34](https://user-images.githubusercontent.com/1557255/217907098-e3d02219-5378-4b1e-98a2-effd1edfbe7f.png)

The game testers mount the intro, data, ending, and outro in that order.

Here's baseline behavior:

![2023-02-09_10-35](https://user-images.githubusercontent.com/1557255/217907549-2ab2f30e-3f7e-4494-9b9b-2cc903dddeb0.png)

![2023-02-09_10-36](https://user-images.githubusercontent.com/1557255/217907547-6494f117-07b8-4e33-99fc-0a3c83aa56c2.png)

(yes, some other bug is blocking the second wildcard set from being handled)

And the PR:

![2023-02-09_10-37](https://user-images.githubusercontent.com/1557255/217907541-4fceb89d-96e0-4d7a-92b9-5d1da23fe9d4.png)

Reply:
> Drive A mounted? Should be the other way around, no?
> 
> ```
> disk1.img, disk2.img, and disk3.img mounted as drive A
> ```

I agree.. maybe they wanted the letter to be shown first (it stands out a bit more).  but I agree, yours is more natural and preferred.

I swapped the drive letter and string:

```c++
MSG_Add("PROGRAM_MOUNT_STATUS_2","Drive %c is mounted as %s\n");

to 

MSG_Add("PROGRAM_MOUNT_STATUS_2","Images %s mounted as drive %c\n");
```

And it segfaults (because now the arguments need to be swapped) in these usage points:

```
src/dos/program_imgmount.cpp
src/dos/program_mount.cpp        
src/dos/program_mount_common.cpp
```

That's not a big deal - but this is worse.. all the foreign strings for `PROGRAM_MOUNT_STATUS_2` need to be reversed:

contrib/resources/translations/utf-8/de.txt::PROGRAM_MOUNT_STATUS_2
contrib/resources/translations/utf-8/en.txt::PROGRAM_MOUNT_STATUS_2
contrib/resources/translations/utf-8/es.txt::PROGRAM_MOUNT_STATUS_2
contrib/resources/translations/utf-8/fr.txt::PROGRAM_MOUNT_STATUS_2
contrib/resources/translations/utf-8/it.txt::PROGRAM_MOUNT_STATUS_2
contrib/resources/translations/utf-8/nl.txt::PROGRAM_MOUNT_STATUS_2
contrib/resources/translations/utf-8/pl.txt::PROGRAM_MOUNT_STATUS_2
contrib/resources/translations/utf-8/ru.txt::PROGRAM_MOUNT_STATUS_2

So I'd either have to hack those strings or ask the translators to fix them (all)? Unfortunately this is feeling too stuck right now :sob:

Reply:
Looks good, I think I slightly misunderstood it then how it works 😄